pkg/monitoring/endpoint.go

```diff
@@ -0,0 +1,307 @@
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	data := e.cli.ExecuteWithJSON(command)
+
+	result := addNodename(&data)
+	writeResponse(result, w)
+}
+
+// ShowBGP returns a result of show bgp command.
+// show bgp (vrf <vrf>) ipv4/ipv6 unicast <input> (longer-prefixes).
+// show bgp vrf <all|vrf> summary.
+func (e *Endpoint) ShowBGP(w http.ResponseWriter, r *http.Request) {
+	vrf := r.URL.Query().Get("vrf")
+	if vrf == "" {
+		vrf = all
+	}
+
+	var data []byte
+	requestType := r.URL.Query().Get("type")
+	switch requestType {
+	case "summary":
+		data = e.cli.ExecuteWithJSON([]string{
+			"show",
+			"bgp",
+			"vrf",
+			vrf,
+			"summary",
+		})
+	case "":
+		protocol := r.URL.Query().Get("protocol")
+		if protocol == "" {
+			protocol = protocolIPv4
+		} else if protocol != protocolIPv4 && protocol != protocolIPv6 {
+			http.Error(w, fmt.Sprintf("protocol '%s' is not supported", protocol), http.StatusBadRequest)
+			return
+		}
+
+		command := []string{
+			"show",
+			"bgp",
+			"vrf",
+			vrf,
+			protocol,
+			"unicast",
+		}
+
+		if err := setInput(r, &command); err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		if err := setLongerPrefixes(r, &command); err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		data = e.cli.ExecuteWithJSON(command)
+	default:
+		http.Error(w, fmt.Sprintf("request of type '%s' is not supported", requestType), http.StatusBadRequest)
+		return
+	}
+
+	result := addNodename(&data)
+	writeResponse(result, w)
+}
+
+// ShowEVPN returns result of show evpn command.
+// show evpn vni json.
+// show evpn rmac vni <all|vrf>.
+// show evpn mac vni <all|vrf>.
+// show evpn next-hops vni <all|vrf> json.
+func (e *Endpoint) ShowEVPN(w http.ResponseWriter, r *http.Request) {
+	var data []byte
+	requestType := r.URL.Query().Get("type")
+	switch requestType {
+	case "":
+		data = e.cli.ExecuteWithJSON([]string{
+			"show",
+			"evpn",
+			"vni",
+		})
+	case "rmac", "mac", "next-hops":
+		vni := r.URL.Query().Get("vni")
+		if vni == "" {
+			vni = all
+		}
+
+		data = e.cli.ExecuteWithJSON([]string{
+			"show",
+			"evpn",
+			requestType,
+			"vni",
+			vni,
+		})
+	default:
+		http.Error(w, fmt.Sprintf("request of type '%s' is not supported", requestType), http.StatusBadRequest)
+		return
+	}
+
+	result := addNodename(&data)
+	writeResponse(result, w)
+}
+
+//+kubebuilder:rbac:groups=core,resources=pods,verbs=list
+
+// PassRequest - when called, will pass the request to all nodes and return their responses.
+func (e *Endpoint) PassRequest(w http.ResponseWriter, r *http.Request) {
+	pods := &corev1.PodList{}
+	matchLabels := &client.MatchingLabels{
+		"app.kubernetes.io/name":      "network-operator",
+		"app.kubernetes.io/component": "worker",
+	}
+
+	err := e.c.List(r.Context(), pods, matchLabels)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("error listing pods: %s", err.Error()), http.StatusInternalServerError)
+		return
+	}
```

Comment:
I am not that happy about the pod discovery, I would probably go for a service / service endpoint based discovery. That way we can pass a servicename as a parameter to the pod.

Reply:
I've changed that. Now the `/all` endpoints will accept parameters `service` and `namespace`, eg.
`/all/show/routes?service=some-svc&namespace=some-namespace`
It will then try to find service and `endpointslices` for this service. If endpointslices are not available it will fallback to the legacy `endpoints`.
I am not sure how this should work with dualstack clusters though. For now I just use the first endpointslices/endpoints object that matches the service.

Reply:
If you want some example of how to the discovery you can read a little bit here :) https://github.com/Cellebyte/kube-latency/blob/master/app.go#L192

Reply:
This is using dns discovery, but as @chdxD1 already stated servicename and namespace could be passed on deployment so it is not needed to have them in the URL :)

Reply:
@Cellebyte Thank you, that's much cleaner. :)
@chdxD1 Service is now configured using `STATUS_SVC_NAME` and `STATUS_SVC_NAMESPACE` variables. Added those to the manager deployment definitions, and added NodePort service `network-operator-status` definition to the deployment.